scaleway/resource_rdb_database.go

```diff
@@ -0,0 +1,160 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/rdb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayRdbDatabase() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayRdbDatabaseCreate,
+		ReadContext:   resourceScalewayRdbDatabaseRead,
+		DeleteContext: resourceScalewayRdbDatabaseDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultRdbInstanceTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validationUUIDorUUIDWithLocality(),
+				Description:  "Instance on which the database is created",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Database name",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"managed": {
+				Type:        schema.TypeBool,
+				Description: "Whether or notre the database is managed",
+				Computed:    true,
+			},
+			"owner": {
+				Type:        schema.TypeString,
+				Description: "User that own the database",
+				Computed:    true,
+			},
+			"size": {
+				Type:        schema.TypeString,
+				Description: "Size of the database",
+				Computed:    true,
+			},
+			// Common
+			"region": regionSchema(),
+		},
+	}
+}
+
+func resourceScalewayRdbDatabaseCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	rdbAPI, region, err := rdbAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	instanceID := d.Get("instance_id").(string)
```

Comment:
region should be read from `instance_id`. `instance_id` should be of the form `region/uuid` like all our ids

Reply:
I'm working on a fix. https://github.com/remyleone/terraform-provider-scaleway/pull/7/

Should I support also non regional id with fallback on the default region?